runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -701,7 +701,17 @@ public synchronized void invalidateLayout() {
             // Don't create a new request for a layout if there is one pending.
             return;
         }
-        layout = fetchLayout(CFUtils.getUninterruptibly(layout).getLayoutServers());
+        List<String> lastKnownLayoutServers = layoutServers;
```

Comment:
Even though we sent it in the fix, this is not correct . The layoutServers list should come from the last know layout. Servers could have moved around but this list is static and this might not be ok.

Reply:
Done. Added a volatile global variable to record the latestLayout seen.